divnix/digga
# develop

- modules: export digga lib in config.lib.digga
- drop builderArgs and pull specialArgs from module
- fix: conflicting declaration error message utility
- !fixup fix: conflicting declaration error message utility
- fix: overlays order - internal, "raked" overlays last
- add test for overlay order
- add module importer (raker)
- bump: `fup` for flake-utils check-utils
- modules: add builds for nixos-generators formats
- flk: add general `build` command, replace iso/doi
- style
- fix: store builds under `./builds`, so .giignore can pick it up.
- the prefix hopeful trick did not work.
- flk: don't set out path for nix build ./result is more obvious for users who've used nix build and then we don't have to worry about file extensions, its already a folder with the right files.
- fix: maybImport to also classic imports that peak for defualt.nix
- nixpkgs-fmt
- flake: drop follows for nixos-generators input
- flake: update lock file should have been included in previous commit to drop follows usage
- flake: update deploy input
- bump fup version (fixup for modules importer)
- fix: maybe import well defined nix modules (with _file)
- ourlib -> config.lib.our
- use config.lib.digga.mkBuild
- mv fullFlake -> example
- add groupByConfig example
- ref: auxiliaries (checks, jobs, devshell)
- fix classicalDevos example
- fmt
- evalnix
- ref: ci to new shift-left testing strategy | please do TDD form now on
- ref: we love the unofficial flakes roadmap
- ref: add proper downstream testing
- fix: check-all
- fix jobs flake
- examples/downstream: use latest digga/develop
- ref: make tests more useful
- imp: checks pass --show-trace flags
- fix: make declaration of suites optional but throw a warning
- ref: tests are becoming a pkgs-less library
- fix: polyfill flake.lock for pure evaluation mode
- fix: have devshell respect it's loaded `nix` version
- imp: conservative digga lib output
- remove code that is no more in active use
- importers: include flattenTree in internal lib
- use deploy's overlay to prevent nixpkgs dep
- fix: add back digga lib checks
- imp: pass down `self` to grouped folder layout
- Revert "use deploy's overlay to prevent nixpkgs dep"
- fix: make nixos.hostDefaults.channelName required
- ref: split digga shell into `divnix/flk` & `divnix/devos/shell` (devshell)
- bump downstream to working tip for tests
- fix: add the entirety of nix#4641 patch
- fix: work arround path:../../ & input.parent error
- imp: reduce flake inputs footprint and increase conciseness
- bump: deploy (with overlay)
- bump: downstream for tests
- zoink: surface regression
- fix: regression
- ref: aclaratory naming
- imp: add portable per aarch users
- fix: checks for portableHomeConfigurations
- imp: better hm user checks naming
- bump: fup (include dynamic output)
- bump: devos for testing
- test: add hm only config
- ref: cleanup hm config generation
- imp: hm user@host idiom should be aligned with host's deployment
- fix: do host-spaced home checks only on that hosts's architecture
- bump: devos for testing
